server/store/mongo_store.go

```diff
@@ -0,0 +1,45 @@
+package store
+
+import (
+	"context"
+	"os"
+	"server/model"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type MongoStore struct {
+	cstr   string
+	client *mongo.Client
+}
+
+const database = "cube"
+const orderCollection = "order"
+
+func NewMongoStore(cstr string) *MongoStore {
+	if cstr == "" {
+		os.Exit(1)
+	}
+	return &MongoStore{cstr: cstr}
+}
+
+func (s *MongoStore) Connect() error {
+	clientOptions := options.Client().ApplyURI(s.cstr).SetDirect(true)
+	c, err := mongo.NewClient(clientOptions)
+	if err != nil {
+		return err
+	}
+	s.client = c
+	err = c.Connect(context.Background())
+	if err != nil {
+		return err
+	}
+	return c.Ping(context.Background(), nil)
+}
+
+func (s *MongoStore) CreateOrder(order *model.Order) error {
```

Comment:
I am not sure whether it is necessary to pass in the context to allow for greater control over request timeouts and cancellations

Reply:
updated